var/spack/repos/builtin/packages/py-poetry-plugin-export/package.py

```diff
@@ -10,11 +10,23 @@ class PyPoetryPluginExport(PythonPackage):
     """Poetry plugin to export the dependencies to various formats"""
 
     homepage = "https://python-poetry.org/"
-    pypi = "poetry-plugin-export/poetry-plugin-export-1.0.7.tar.gz"
+    pypi = "poetry-plugin-export/poetry_plugin_export-1.6.0.tar.gz"
 
+    version("1.6.0", sha256="091939434984267a91abf2f916a26b00cff4eee8da63ec2a24ba4b17cf969a59")
     version("1.0.7", sha256="f6ac707ae227b06b2481249ed2678ff6b810b3487cac0fbb66eb0dc2bfd6ecf1")
 
+    depends_on("python@3.8:3", when="@1.6:", type=("build", "run"))
     depends_on("python@3.7:3", type=("build", "run"))
-    depends_on("py-poetry-core@1.1:1", type=("build", "run"))
+    depends_on("py-poetry-core@1.7:1", when="@1.6", type=("build", "run"))
+    depends_on("py-poetry-core@1.1:1", when="@1.0", type=("build", "run"))
+    depends_on("py-poetry@1.6:1", when="@1.6", type=("build", "run"))
 
-    # depends_on("py-poetry@1.2:1", type="run") # circular dependency
```

Comment:
Isn't this circular dependency still an issue?

Reply:
@alalazo poetry has a run-dep on poetry-plugin-export, and poetry-plugin-export has a run-dep on poetry. Would love to be able to model this correctly in the concretizer someday.